site/en/blog/wordpress-playground/index.md

```diff
@@ -0,0 +1,236 @@
+example, `php_7_3.wasm` instead of, say, `php_8_2.wasm`.
+
+### Database is supported with a SQL translation layer
+
+WordPress requires MySQL. However, there isn't a WebAssembly version of MySQL
+you could run in the browser. WordPress Playground therefore ships PHP with the
+[native SQLite driver](https://www.php.net/manual/en/ref.pdo-sqlite.php) and
+leans on SQLite.
+
+But how can WordPress run on a different database?
+
+Behind the scenes, the official
+[SQLite Database Integration](https://github.com/WordPress/sqlite-database-integration)
+plugin intercepts all MySQL queries and rewrites them in SQLite dialect. The 2.0
+release ships
+[a new WordPress Playground-informed translation layer](https://github.com/WordPress/sqlite-database-integration/pull/9)
+that allows WordPress on SQLite to pass 99% of the WordPress unit test suite.
+
+### The web server lives inside the browser
+
+In a regular WordPress, clicking on a link, say _Blog,_ would initiate an HTTP
+request to the remote backend to fetch the `blog` page. However, WordPress
+Playground has no remote backend. It has a
+[Service Worker](https://developer.mozilla.org/docs/Web/API/Service_Worker_API)
+that intercepts all the outgoing requests and passes them to an in-browser PHP
+instance running in a separate
+[Web Worker](https://developer.mozilla.org/docs/Web/API/Web_Workers_API/Using_web_workers).
+
+{% Img src="image/8WbTDNrhLsU0El80frMBGE4eMCD3/EAYGDJ5cBGHFnQuFYjEw.png", alt="Flow diagram starting with an iframe pointing at the resource wp-admin, calls to which are intercepted by the service worker, rendered in the worker thread, and ultimately translated to a WordPress response by the in-browser server.", width="800", height="675" %}
+
+### Networking is supported through WebSockets
+
+When it comes to networking, WebAssembly programs are limited to calling
+JavaScript APIs. It is a safety feature, but also presents a challenge. How do
+you support low-level, synchronous networking code used by PHP with the
+high-level asynchronous APIs available in JavaScript?
+
+For WordPress Playground, the answer involves a WebSocket to TCP socket proxy,
+[Asyncify](https://emscripten.org/docs/porting/asyncify.html), and patching deep
+PHP internals like `php_select`. It's complex, but there's a reward. The
+Node.js-targeted PHP build can request web APIs, install composer packages, and
+even connect to a MySQL server.
+
+## Since it's just WebAssembly, WordPress can be used in even more places than the browser
+
+Since WordPress can now run on WebAssembly, it can actually be used in even more
+places than the browser. A Node.js server can now run WebAssembly WordPress
+since it's still the same V8 engine. Of course with StackBlitz you can also run
+Node.js directly in the browser, meaning that you could run WordPress and PHP
+compiled to WebAssembly, executing in Node.js, which is also compiled to
+WebAssembly running
```

Comment:
One last suggestion:

```suggestion
## WordPress can be used in even more places than the browser

Since WordPress can now run on WebAssembly, you could also run it in a Node.js
server – it's the same V8 engine! Of course with StackBlitz you can also run
Node.js directly in the browser, meaning that you could run WordPress and PHP
compiled to WebAssembly, executing in Node.js, which is also compiled to
WebAssembly running
```